### DISCUSSION TITLE
Criando um Array numa função e retornando a soma de seus elementos

### DISCUSSION THREAD

**Discussion by leomalm** [frontendbr/forum#2395] [Perguntas e Respostas]
Fala, pessoal! A pergunta é basicamente o que está no título. Criei um Array de números múltiplos de 3 ou 5 menores que um número qualquer informado pelo usuário. Esse número é incluído em uma função que retorna o Array. Porém estou com dificuldade para entregar a soma dos elementos desse Array.

Venho encontrando na internet a proposta recorrente de definir uma variável var soma = 0; e depois aplicar um For que somaria os itens do Array, iterando a variável soma, mas não consegui integrar essa solução no meu código.

Conseguem me ajudar?

**Comment by JoaoVictor6** [comment]
Olá! Que tal usar o metodo [Reduce](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)?

Seria algo como:
```js
const sum = arr.reduce(
 (prevValue, currentValue) => prevValue+currentValue, 
  0
)
```

**Reply by leomalm** [reply]
Oi, João! Obrigado pela dica.

Os termos prevValue e currentValue seriam nativos da função reduce?

**Reply by jampow** [reply]
São apenas nomes que ele escolheu dar, poderia ser `a`e `b`, ou (acho bem comum) chamar o primeiro item de `acc` (accumulator), já que o valor retornado é "acumulado" nesta variável. O `0` no final é a inicialização do `acc`

```
const sum = arr.reduce(
 (acc, item) => acc + item, 
  0
)
```

**Comment by censuradho** [comment]
Como o joão já demonstrou, da pra utilizar o reducer, deixo aqui um link para uma documentação na [MDN](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) explicando certinho como utilizar esse metodo:
